tests/functional/adapter/custom_schema_tests/test_custom_database.py

```diff
@@ -0,0 +1,111 @@
+import pytest
+import os
+from dbt.tests.util import (
+    check_relations_equal,
+    check_table_does_exist,
+    run_dbt
+)
+from tests.functional.adapter.custom_schema_tests.seeds import seed_queries
+
+
+_VIEW_1_SQL = """
+select * from {{ target.schema }}.seed
+""".lstrip()
+
+_VIEW_2_SQL = """
+{{ config(database='alt') }}
+select * from {{ ref('view_1') }}
+""".lstrip()
+
+_VIEW_3_SQL = """
+{{ config(database='alt', materialized='table') }}
+
+
+with v1 as (
+
+    select * from {{ ref('view_1') }}
+
+),
+
+v2 as (
+
+    select * from {{ ref('view_2') }}
+
+),
+
+combined as (
+
+    select last_name from v1
+    union all
+    select last_name from v2
+
+)
+
+select
+    last_name,
+    count(*) as count
+
+from combined
+group by 1
+""".lstrip()
+
+_CUSTOM_DB_SQL = """
+{% macro generate_database_name(database_name, node) %}
+    {% if database_name == 'alt' %}
+        {{ env_var('SNOWFLAKE_TEST_ALT_DATABASE') }}
+    {% elif database_name %}
+        {{ database_name }}
+    {% else %}
+        {{ target.database }}
+    {% endif %}
+{% endmacro %}
+""".lstrip()
+
+ALT_DATABASE = os.getenv("SNOWFLAKE_TEST_ALT_DATABASE")
+
+class TestOverrideDatabase:
+    @pytest.fixture(scope="class", autouse=True)
```

Comment:
There's only one test method in this class, so it doesn't currently matter. But the starting data appears to be written in a way that suggests it should be refreshed for every test, not every class. So the scope should probably be "function" instead of "class", in the event we ever add a second test.

Reply:
Followed this convention: [setUp](https://github.com/dbt-labs/dbt-core/blob/ccb4fa26cd682812f15d2fd50b37a8c422e092f8/tests/adapter/dbt/tests/adapter/simple_seed/test_seed.py#L45)
If I change it to `function` it doesn't pass when run in parallel.

Reply:
Added in the next commit.